core/templates/services/audio-player.service.ts

```diff
@@ -16,162 +16,235 @@
+    if (this._currentTrack) {
+      const currentSeconds = this._currentTrack.seek();
+      if (typeof currentSeconds !== 'number') {
+        return;
       }
-    };
-
-    var _forward = function(seconds) {
-      if (_currentTrack) {
-        var currentSeconds = _currentTrack.progress * _currentTrack.duration;
-        if (currentSeconds + seconds < _currentTrack.duration) {
-          var forwardedProgress = (
-            (currentSeconds + seconds) / _currentTrack.duration);
-          _currentTrack.progress = forwardedProgress;
-        }
+      if (currentSeconds + seconds < this._currentTrack.duration()) {
+        this._currentTrack.seek(currentSeconds + seconds);
       }
-    };
+    }
+  }
 
-    return {
-      load: function(filename) {
-        return $q(function(resolve, reject) {
-          _load(filename, resolve, reject);
-        });
-      },
-      play: function() {
-        _play();
-      },
-      pause: function() {
-        _pause();
-      },
-      stop: function() {
-        _stop();
-      },
-      rewind: function(seconds) {
-        _rewind(seconds);
-      },
-      forward: function(seconds) {
-        _forward(seconds);
-      },
-      getCurrentTime: function() {
-        if (_currentTrack) {
-          return Math.round(_currentTrack.currentTime);
-        } else {
-          return 0;
-        }
-      },
-      getAudioDuration: function() {
-        if (_currentTrack && _currentTrack.audio) {
-          return Math.round(_currentTrack.audio.duration);
-        } else {
-          return 0;
-        }
-      },
-      getProgress: function() {
-        if (!_currentTrack) {
-          return 0;
-        }
-        return _currentTrack.progress;
-      },
-      setProgress: function(progress) {
-        if (_currentTrack) {
-          _currentTrack.progress = progress;
-        }
-      },
-      isPlaying: function() {
-        return Boolean(_currentTrack && !_currentTrack.paused);
-      },
-      isTrackLoaded: function() {
-        return Boolean(_currentTrack);
-      },
-      clear: function() {
-        _currentTrack = null;
-        _currentTrackFilename = null;
-      },
-      get onAutoplayAudio() {
-        return _autoplayAudioEventEmitter;
+  load(filename: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this._load(filename, resolve, reject);
+    });
+  }
+
+  play(): void {
+    this._play();
+  }
+
+  pause(): void {
+    this._pause();
+  }
+
+  stop(): void {
+    this._stop();
+  }
+
+  rewind(seconds: number): void {
+    this._rewind(seconds);
+  }
+
+  forward(seconds: number): void {
+    this._forward(seconds);
+  }
+
+  getCurrentTime(): number {
+    if (this._currentTrack) {
+      const sec = this._currentTrack.seek();
+      if (typeof sec !== 'number') {
+        return 0;
       }
-    };
+      return Math.floor(sec);
+    } else {
+      return 0;
+    }
+  }
+
+  setCurrentTime(val: number): void {
```

Comment:
why are we allowing to set time through this service? (I do see gets/setProgress function will that be sufficient?)

Reply:
>  I have kept both for now in case someone wants to use the old one... for backward compatibility.

We should avoid having unused code and as it's an internal code and not a public API so we should expect developers to re-write the function if needed instead of having a dead function! wdyt?

Reply:
Removed both functions